src/eval-frame/reducers/eval-frame-reducer.js

```diff
@@ -43,13 +47,12 @@ const notebookReducer = (state = newNotebook(), action) => {
     case 'UPDATE_EVAL_FRAME_FROM_INITIAL_JSMD': {
       const cells = action.stateUpdatesFromEditor.cells
         .map(c => Object.assign(newCell(), c))
-      console.log('cells cells cells', cells)
       const newState = Object.assign(
         newNotebook(),
         action.stateUpdatesFromEditor,
         { cells },
+        { userDefinedVarNames: getUserDefinedVarsFromWindow() },
```

Comment:
architectural comment: since we have started being more careful about keeping impure side-effect-y functions like `getUserDefinedVarsFromWindow()` in action creators rather than the reducer, i think this would be a case in which we should follow that practice.

(just added issue #995 to track this general cleanup)

Reply:
`getUserDefinedVarsFromWindow` is not impure, though from what I'm understanding you want `userDefinedVarNames` to be built via an action creator and not collected from `window` in a reducer, correct?

Reply:
on second thought I see your point. Will circle back on this tomorrow